src/macros.jl

```diff
@@ -873,34 +873,45 @@ esc_nonconstant(x::Number) = x
 esc_nonconstant(x::Expr) = isexpr(x,:quote) ? x : esc(x)
 esc_nonconstant(x) = esc(x)
 
+mutable struct VariableInfo{S, T, U, V}
+    haslb::Bool
+    lowerbound::S
+    hasub::Bool
+    upperbound::T
+    hasfix::Bool
+    fixedvalue::U
+    binary::Bool
+    integer::Bool
+    hasstart::Bool
+    start::V
+end
+
 # Returns the type of what `constructvariable!` would return with these starting positional arguments.
 variabletype(m::Model) = Variable
 # Returns a new variable belonging to the model `m`. Additional positional arguments can be used to dispatch the call to a different method.
 # The return type should only depends on the positional arguments for `variabletype` to make sense.
-function constructvariable!(m::Model, _error::Function, haslb::Bool, lowerbound::Number, hasub::Bool, upperbound::Number,
-                            hasfix::Bool, fixedvalue::Number, binary::Bool, integer::Bool, name::String,
-                            hasstart::Bool, start::Number; extra_kwargs...)
+function constructvariable!(m::Model, _error::Function, info::VariableInfo, name::String; extra_kwargs...)
```

Comment:
This is much better. What's the motivation for leaving `name` out of `VariableInfo`?